ui/src/main/webapp/app/gaffer/results-service.js

```diff
@@ -60,17 +60,38 @@ angular.module('app').factory('results', ['events', function(events) {
                 if(result.class === "uk.gov.gchq.gaffer.data.element.Entity") {
                     if(result.vertex !== undefined && result.vertex !== '') {
                         incomingResults.entities.push(result);
-                        results.entities.push(result);
+                        var toAdd = true;
+                        results.entities.forEach(function(element) {
```

Comment:
Could this be made more efficient? Iterating through the results for each new element could soon become slow with 1000+ complex elements.

Reply:
I tried looking at the common.pushIfUnique commands however the only applicable choice required a value as a key and most of these result objects don't have a unique 'id'. I've opted to use arrays.some() to check if the array contains the element.

Reply:
Will an array.indexOf() lookup work?

Reply:
Gave it a quick go but doesn't seem to work using array.indexOf(element) !== -1. Receiving duplicate results.

Reply:
Ah okay. Fair enough